### DISCUSSION TITLE
There is a color difference in the non_stroking_color

### DISCUSSION THREAD

**Discussion by KaboChow** [jsvine/pdfplumber#1215] [Q&A]
Hi Team,
I am trying to use the following JS code to convert the obtained non_stoking.color into RGB color for use in HTML.
`
              
                if(fillColor.length===1){
                    return 'rgb(' + (255*(1.0-fillColor[0])).toFixed(0) + ')';
                }
                
                if(fillColor.length===3){
                    const r = 255 * fillColor[0];
                    const g = 255 * fillColor[1];
                    const b = 255 * fillColor[2];
                    return 'rgb(' + r.toFixed(0) + ',' + g.toFixed(0) +',' + b.toFixed(0) + ')';
                }
               
                if(fillColor.length===4){
                    const r = 255*(1.0-(fillColor[0]+fillColor[3]));
                    const g = 255*(1.0-(fillColor[1]+fillColor[3]));
                    const b = 255*(1.0-(fillColor[2]+fillColor[3]));
                    return 'rgb(' + r.toFixed(0) + ',' + g.toFixed(0) +',' + b.toFixed(0) + ')';
                }
`

But there are obvious color differences in the obtained colors.
pdf:
![image](https://github.com/user-attachments/assets/f9b23bf4-5f46-4fbc-9304-ecaeb5ce2415)
html:
![image](https://github.com/user-attachments/assets/759684c0-56ab-4ca7-a481-2e9d1337590b)

This is the PDF I used for testing:
[TEST 5 - 48in-Height-Chef-Supplies-Banner-d4-ced6f345009972ad3e50309422be6654.pdf](https://github.com/user-attachments/files/17178323/TEST.5.-.48in-Height-Chef-Supplies-Banner-d4-ced6f345009972ad3e50309422be6654.pdf)

Does anyone know how to solve it? thank you

**Comment by jsvine** [comment]
Given that this doesn't seem to be a bug in `pdfplumber`, I'm converting this to a discussion. The issue here seems to be that the colors are defined as CMYK instead of RGB(A). Plugging in the color values for the `CHEF` curves, you get a color that looks correct:

<img width="463" alt="Screenshot 2024-10-17 at 9 38 01 PM" src="https://github.com/user-attachments/assets/1817379e-cc7a-436f-b9ce-f8b7a2e407e6">

**Comment by KaboChow** [comment]
Hi! @jsvine, thank you for your reply!

You're right, when the curve uses the CMYK format, the color is normal!
I apologize for not fully expressing my question.

When the curve uses the RGB format, the value of non_stroking_color shows the following issues:

The correct color is purple, but the value is [1].
There are cases where negative values appear, such as [76.470588, -3, -8].
I am unable to distinguish whether non_stroking_color is using CMYK or RGB based on the limited information.

This is the PDF I used for testing:
[TEST 62 - trouse-fd7b74be16be80f5c0397ef382664f74.pdf](https://github.com/user-attachments/files/17427156/TEST.62.-.trouse-fd7b74be16be80f5c0397ef382664f74.pdf)
[TEST 66 - IE_Logo_Vector-2dc8c79f1cea26d809275d24b699b7c0.pdf](https://github.com/user-attachments/files/17427158/TEST.66.-.IE_Logo_Vector-2dc8c79f1cea26d809275d24b699b7c0.pdf)

**Comment by mkl-public** [comment]
[TEST 62 - trouse-fd7b74be16be80f5c0397ef382664f74.pdf](https://github.com/user-attachments/files/17427156/TEST.62.-.trouse-fd7b74be16be80f5c0397ef382664f74.pdf): Here a **Lab** color space with three components is used. The latter two components may be negative.

[TEST 66 - IE_Logo_Vector-2dc8c79f1cea26d809275d24b699b7c0.pdf](https://github.com/user-attachments/files/17427158/TEST.66.-.IE_Logo_Vector-2dc8c79f1cea26d809275d24b699b7c0.pdf): Here **Separation** color spaces with a single component are used.
